### PR TITLE
feat: extensions

### PR DESCRIPTION
**Pull request by joshuasing**
This pull request adds an Extension system to Chameleon as a replacement for the half-broken module system.

### CODE REVIEW

**Review by SLLCoding — APPROVED** [review]

**Review by SLLCoding — COMMENTED** [review]
Are the extensions' #onEnable methods called when post loaded?

**Comment by joshuasing** [comment]
> Are the extensions' #onEnable methods called when post loaded?

Doesn't look like it, I'll fix that and test it before merging

File: build-logic/src/main/kotlin/chameleon.common.gradle.kts [hunk 1]
```diff
@@ -23,4 +23,17 @@
...
     id("chameleon.publishing")
+}
+
+val libs = extensions.getByType<VersionCatalogsExtension>().named("libs")
+
+dependencies {
+    testImplementation(platform(libs.findLibrary("test-junit-bom").get()))
+    testImplementation(libs.findLibrary("test-junit-api").get())
+    testImplementation(libs.findLibrary("test-junit-engine").get())
+    testImplementation(libs.findLibrary("test-junit-params").get())
```
Comment (SLLCoding) [inline build-logic/src/main/kotlin/chameleon.common.gradle.kts:34]: Heh?

**Review by SLLCoding — APPROVED** [review]

File: build-logic/src/main/kotlin/chameleon.common.gradle.kts (hunk 1 above)
Comment (joshuasing) [reply build-logic/src/main/kotlin/chameleon.common.gradle.kts:34]: This allows any module using `chameleon.common` to use JUnit for tests

**Pull request merged**